saidump/saidump.cpp

```diff
@@ -399,7 +417,170 @@ void dumpGraph(const TableDump& td)
     std::cout << "}" << std::endl;
 }
 
-int main(int argc, char ** argv)
+/*
+ preprocess the input json file to make sure it's a valid json file.
+*/
+static int preProcessFile(std::string file_name)
```

Comment:
if you want to make comment like this, make it like doxygen format
```c++
/**
 * @brief Preprocess ..
 * /
```

Reply:
```c++
static int preProcessFile(const std::string& fileName)
```

Reply:
return value should be sai_status_t or bool sincd you are using this values

Reply:
Fix it.